syntax = "proto3";
package CAS;
service AuthAndRegistService{
    rpc TryRegistr(RegistrationInput) returns (RegistrationResult);
    rpc TryVerifRegistr(RegistrationVerificationInput) returns (RegistrationVerificationResult);
    rpc Authenticate(AuthInput) returns (AuthResult);
    rpc AuthFromPassword(PasswordAuthInput) returns (PasswordAuthResult);
}
message RegistrationInput{
    string email = 1;
    string password = 2;
}
message RegistrationVerificationInput{
    string randomDataToken = 1;
}
message AuthInput{
    string jwtToken = 1;
}

message PasswordAuthInput{
	string email = 1;
	string password = 2;
}
message RegistrationResult{
    string answer = 1;
    bool isOk = 2;
}

message RegistrationVerificationResult{
    string response_message = 1;
}
message AuthResult{
    int64 userId = 1;
    string nextToken = 2;
}
message PasswordAuthResult{
	string jwtToken = 1;
	int64 user_id = 2;
}

